include/sound/soc-acpi.h

```diff
@@ -75,18 +75,47 @@ struct snd_soc_acpi_mach_params {
 };
 
 /**
- * snd_soc_acpi_link_adr: ACPI-based list of _ADR, with a variable
- * number of devices per link
+ * snd_soc_acpi_endpoint: endpoint descriptor
+ *
+ * @num: endpoint number (mandatory, unique per device)
+ * @aggregated: 0 (independent) or 1 (logically grouped)
+ * @group_position: zero-based order (only when @aggregated is 1)
+ * @group_id: platform-unique group identifier (only when @aggregrated is 1)
+ */
+struct snd_soc_acpi_endpoint {
+	u8 num;
+	u8 aggregated;
+	u8 group_position;
+	u8 group_id;
+};
+
+/**
+ * snd_soc_acpi_adr_device: descriptor for device, include _ADR
+ * reresented at 64 bits and endpoint information
+ *
+ * @adr: 64 bit ACPI _ADR value
+ * @num_endpoints: number of endpoints for this device
+ * @endpoints: array of endpoints
+ */
+struct snd_soc_acpi_adr_device {
+	const u64 adr;
+	const u8 num_endpoints;
+	const struct snd_soc_acpi_endpoint *endpoints;
+};
+
+/**
+ * snd_soc_acpi_link_adr: ACPI-based list of devices identified with an
+ * _ADR, with a variable number of devices per link
```

Comment:
I'm not sure about this: I think usually the "short description" should be one line, does it still work with more than that? You can try to generate some documentation from this to check. Or just go with 1 line for safety and put any details below under all the struct members. Also I think this first line usually has the format of " * name - description" - not with a semicolon, but I'm not sure how hard that rule is either.

Reply:
will check, thanks for spotting this. I ran make W=1 which checks the kernel doc and didn't see any warning, but it doesn't hurt to check the documentation itself

Reply:
fixed with the proper format, thanks @lyakh